app/v2/controllers/V2DeparturesController.scala

```diff
@@ -70,26 +75,56 @@ class V2DeparturesControllerImpl @Inject() (
   def submitDeclaration(): Action[Source[ByteString, _]] =
     (authActionNewEnrolmentOnly andThen messageSizeAction()).async(streamFromMemory) {
       implicit request =>
-        withTemporaryFile {
-          (temporaryFile, source) =>
-            logger.info("Version 2 of endpoint has been called")
-
-            implicit val hc: HeaderCarrier = HeaderCarrierConverter.fromRequest(request)
-
-            (for {
-              _ <- validationService.validateXML(MessageType.DepartureDeclaration, source).asPresentation
-
-              fileSource = FileIO.fromPath(temporaryFile)
-
-              declarationResult <- departuresService.saveDeclaration(request.eoriNumber, fileSource).asPresentation
-              _ <- routerService
-                .send(MessageType.DepartureDeclaration, request.eoriNumber, declarationResult.departureId, declarationResult.messageId, fileSource)
-                .asPresentation
-            } yield declarationResult).fold[Result](
-              presentationError => Status(presentationError.code.statusCode)(Json.toJson(presentationError)),
-              result => Accepted(HateoasDepartureDeclarationResponse(result.departureId))
-            )
-        }.toResult
+        request.headers.get("Content-Type") match {
+          case Some("application/xml")  => submitDeclarationXML
+          case Some("application/json") => submitDeclarationJSON
+          case contentType              => handleInvalidContentType(contentType)
+        }
+
     }
 
+  def submitDeclarationJSON(implicit request: AuthenticatedRequest[Source[ByteString, _]]): Future[Result] = {
+    request.body.to(Sink.ignore).run()
+    Future.successful(Accepted)
+  }
+
+  def submitDeclarationXML(implicit request: AuthenticatedRequest[Source[ByteString, _]]): Future[Result] =
+    withTemporaryFile {
+      (temporaryFile, source) =>
+        logger.info("Version 2 of endpoint has been called")
```

Comment:
We probably don't want to keep this logging message long term, but for now, perhaps we should either change this to say "XML", or we should move this to the `submitDeclaration` method.